testing/trino-testing/src/main/java/io/trino/testing/PlanDeterminismChecker.java

```diff
@@ -49,13 +48,15 @@ public void checkPlanIsDeterministic(String sql)
 
     public void checkPlanIsDeterministic(Session session, String sql)
     {
-        IntStream.range(1, MINIMUM_SUBSEQUENT_SAME_PLANS)
-                .mapToObj(attempt -> getPlanText(session, sql))
-                .map(planEquivalenceFunction)
-                .reduce((previous, current) -> {
-                    assertEquals(previous, current);
-                    return current;
-                });
+        String previous = null;
+        for (int attempt = 1; attempt < MINIMUM_SUBSEQUENT_SAME_PLANS; attempt++) {
```

Comment:
We're checking that MINIMUM_SUBSEQUENT_SAME_PLANS-1 plans are the same, aren't we?

this is pre-existing, but since you're rewriting it...

Reply:
Yes, this is something that bothered me, too. Looks like an off-by-one to me, so I wanted a second opinion of someone who can judge the intent more precisely (see https://github.com/trinodb/trino/pull/6879/files#r576112688). I'll "fix" it and see what people think (though a regression test would be nice, too)

Reply:
regression test is not applcable -- this is just assertion coide in a test; there is no way to test that (other than contrived test cases which wouldn't be useful)

Reply:
Yeah, I figured as much. It's just making the existing tests a little stronger than before